spec/services/container_deployment_service_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'recursive-open-struct'
+
+describe ContainerDeploymentService do
+  before do
+    %w(amazon openstack google azure redhat vmware).each do |p|
+      network = FactoryGirl.create(:network, :ipaddress => "127.0.0.1")
+      @hardware = FactoryGirl.create(:hardware)
+      @hardware.networks << network
```

Comment:
this creates 6 Network and Hardware pairs but only uses the last?  move out of the loop.